misc/config_tools/xforms/vm_configurations.c.xsl

```diff
@@ -238,18 +239,24 @@
       <xsl:value-of select="acrn:ifdef('VM0_PASSTHROUGH_TPM')" />
       <xsl:value-of select="acrn:initializer('pt_tpm2', 'true')" />
       <xsl:value-of select="acrn:initializer('mmiodevs[0]', '{', true())" />
-      <xsl:value-of select="acrn:initializer('user_vm_pa', 'VM0_TPM_BUFFER_BASE_ADDR_GPA')" />
-      <xsl:value-of select="acrn:initializer('service_vm_pa', 'VM0_TPM_BUFFER_BASE_ADDR')" />
-      <xsl:value-of select="acrn:initializer('size', 'VM0_TPM_BUFFER_SIZE')" />
+      <xsl:value-of select="acrn:initializer('name', concat($quot, 'tpm2', $quot))" />
+      <xsl:value-of select="acrn:initializer('res[0].base_gpa', 'VM0_TPM_BUFFER_BASE_ADDR_GPA')" />
+      <xsl:value-of select="acrn:initializer('res[0].base_hpa', 'VM0_TPM_BUFFER_BASE_ADDR')" />
+      <xsl:value-of select="acrn:initializer('res[0].size', 'VM0_TPM_BUFFER_SIZE')" />
+      <xsl:if test="//capability[@id='log_area']">
+        <xsl:value-of select="acrn:initializer('res[1].base_gpa', 'VM0_TPM_EVENTLOG_BASE_ADDR')" />
+        <xsl:value-of select="acrn:initializer('res[1].base_hpa', 'VM0_TPM_EVENTLOG_BASE_ADDR')" />
+        <xsl:value-of select="acrn:initializer('res[1].size', 'VM0_TPM_EVENTLOG_SIZE')" />
+      </xsl:if>
       <xsl:text>},</xsl:text>
       <xsl:value-of select="$newline" />
       <xsl:value-of select="$endif" />
       <xsl:value-of select="acrn:ifdef('P2SB_BAR_ADDR')" />
       <xsl:value-of select="acrn:initializer('pt_p2sb_bar', 'true')" />
       <xsl:value-of select="acrn:initializer('mmiodevs[0]', '{', true())" />
-      <xsl:value-of select="acrn:initializer('user_vm_pa', 'P2SB_BAR_ADDR_GPA')" />
-      <xsl:value-of select="acrn:initializer('service_vm_pa', 'P2SB_BAR_ADDR')" />
-      <xsl:value-of select="acrn:initializer('size', 'P2SB_BAR_SIZE')" />
+      <xsl:value-of select="acrn:initializer('res[0].base_gpa', 'P2SB_BAR_ADDR_GPA')" />
```

Comment:
please rename base_gpa to user_vm_pa and base_hpa to host_hpa

Reply:
Also, please config CONFIG_SECURITY_VM_FIXUP for the platform which needs to do TPM2 fixup, and config the VM flag GUEST_FLAG_TPM2_FIXUP for the VM which needs to do TPM2 fixup.